python/ray/dataframe/dataframe.py

```diff
@@ -2774,37 +2773,66 @@ def quantile(self, q=0.5, axis=0, numeric_only=True,
                     are the quantiles.
         """
 
-        def quantile_helper(df, q, axis, numeric_only, interpolation):
-            try:
-                return df.quantile(q=q, axis=axis, numeric_only=numeric_only,
-                                   interpolation=interpolation)
-            except ValueError:
-                return pd.Series()
+        # deal with error checking for bad dtyping when numeric_only is False
+        if not numeric_only:
+            # check if there are any object columns
+            for t in self.dtypes:
+                if t == np.dtype('O'):
+                    e_msg = "Cannot call quantile on columns of type " + str(t)
+                    raise TypeError(e_msg)
+
+        # check that all qs are between 0 and 1
+        msg = ("percentiles should all be in the interval [0, 1]. "
```

Comment:
Use the pandas `_check_percentile` method here.

Reply:
Updated